tableauserverclient/models/task_item.py

```diff
@@ -1,39 +1,52 @@
 import xml.etree.ElementTree as ET
 from .target import Target
+from .schedule_item import ScheduleItem
+from ..datetime_helpers import parse_datetime
 
 
 class TaskItem(object):
-    def __init__(self, id_, task_type, priority, consecutive_failed_count=0, schedule_id=None, target=None):
+    class Type:
+        ExtractRefresh = "extractRefresh"
+        MaterializeViews = "materializeViews"
+
+    def __init__(self, id_, task_type, priority, consecutive_failed_count=0, schedule_id=None,
+                 schedule_item=None, last_run_at=None, target=None):
         self.id = id_
         self.task_type = task_type
         self.priority = priority
         self.consecutive_failed_count = consecutive_failed_count
         self.schedule_id = schedule_id
+        self.schedule_item = schedule_item
+        self.last_run_at = last_run_at
         self.target = target
 
     def __repr__(self):
         return "<Task#{id} {task_type} pri({priority}) failed({consecutive_failed_count}) schedule_id({" \
                "schedule_id}) target({target})>".format(**self.__dict__)
 
     @classmethod
-    def from_response(cls, xml, ns):
+    def from_response(cls, xml, ns, task_type=Type.ExtractRefresh):
         parsed_response = ET.fromstring(xml)
         all_tasks_xml = parsed_response.findall(
-            './/t:task/t:extractRefresh', namespaces=ns)
+            './/t:task/t:{}'.format(task_type), namespaces=ns)
 
         all_tasks = (TaskItem._parse_element(x, ns) for x in all_tasks_xml)
 
         return list(all_tasks)
 
     @classmethod
     def _parse_element(cls, element, ns):
-        schedule = None
+        schedule_id = None
+        schedule_item = None
         target = None
-        schedule_element = element.find('.//t:schedule', namespaces=ns)
+        last_run_at = None
         workbook_element = element.find('.//t:workbook', namespaces=ns)
         datasource_element = element.find('.//t:datasource', namespaces=ns)
-        if schedule_element is not None:
-            schedule = schedule_element.get('id', None)
+        last_run_at_element = element.find('.//t:lastRunAt', namespaces=ns)
+
+        schedule_item_list = ScheduleItem.from_element(element, ns)
+        if len(schedule_item_list) >= 1:
```

Comment:
Just out of curiosity, can a task ever return more than 1 schedule?

Reply:
No, a task can have only one schedule. Here is an example: https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref.htm#add_workbook_to_schedule